### PR TITLE
Use GitHub Actions to validate inventory

### LINKED ISSUE #8: Validate the inventory state on CI
It would be great if we could have `sr inv-validate` run on CI in order to pick up on state issues. Since the primary mode of interaction with the inventory is direct-push rather than PRs, this isn't perfect, though feels like a useful starting point.

### PR DESCRIPTION
**Pull request by RealOrangeOne**
Fixes #8 

First trial of GitHub actions - It's pretty nice!

### CODE REVIEW

**Comment by sedders123** [comment]
I am a bit surprised having a `.github` folder is valid with the inventory given how strict it is regarding normal files. I suppose its ignored as its prefixed with a `.`

File: .github/workflows/validate-inventory.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    steps:
+    - uses: actions/checkout@v1
+    - name: Set up Python 3.7
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.7
+    - name: Install SR tools
+      run: |
+        python -m pip install --upgrade pip
+        pip install sr.tools
```
Comment (trickeydan) [inline .github/workflows/validate-inventory.yml]: Is it possible to cache this install? I'm not familiar enough with GHA to know what runtime it uses.

**Comment by trickeydan** [comment]
Nice :tada:

**Comment by trickeydan** [comment]
Question / Meta: How does this affect users? Do they see it? The commits are pushed directly to master, so I don't see how they would be blocked if the state is invalid.

File: .github/workflows/validate-inventory.yml (hunk 1 above)
Comment (RealOrangeOne) [reply .github/workflows/validate-inventory.yml]: You cannot, which is unfortunate. It's possible it'll be added in a future version, but the build is only ~30s, which is fine in this situation.

**Comment by RealOrangeOne** [comment]
> How does this affect users?

Simple, it doesn't. Blocking pushes based on this is something better enforced by training, as it's currently not supported. This does however allow us to quickly see if the inventory is valid, find out the commit which broke it, and hopefully rectify it.

I did try adding a README to the repo with a badge (569f0e94f49ab49f3adf1d72b8cb462ab445f858),but this yielded an invalid inventory. @PeterJCLaw is there any way we can change things such that certain directories are completely ignored from the inventory?

**Comment by sedders123** [comment]
>  @PeterJCLaw is there any way we can change things such that certain directories are completely ignored from the inventory?

I think this PR allows READMEs https://github.com/srobo/tools/pull/24

File: .github/workflows/validate-inventory.yml (hunk 1 above)
Comment (sedders123) [reply .github/workflows/validate-inventory.yml]: Support for caching has been [discussed](https://github.com/actions/toolkit/issues/47#issuecomment-521108865) and is apparently "on the feature backlog short list".

**Comment by sedders123** [comment]
Any reason we haven't merged this in? Not sure a README is necessary for this

File: .github/workflows/validate-inventory.yml (hunk 1 above)
Comment (PeterJCLaw) [reply .github/workflows/validate-inventory.yml]: https://github.com/actions/cache seems to have since appeared.

**Pull request merged**